### PR TITLE
chore(hlapi): move integrations tests into src dir

### PR DESCRIPTION
**Pull request by tmontaigu**
Fixes https://github.com/zama-ai/tfhe-rs-internal/issues/220

### CODE REVIEW

File: tfhe/src/high_level_api/shortints/tests.rs [hunk 1]
```diff
@@ -1,8 +1,9 @@
+#![allow(clippy::assign_op_pattern)]
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/shortints/tests.rs:1]: what is this for ?

Comment (tmontaigu) [reply tfhe/src/high_level_api/shortints/tests.rs:1]: clippy conplains about doing `let c = a + b`, it would like us to do `a+=b;`

Comment (IceTDrinker) [reply tfhe/src/high_level_api/shortints/tests.rs:1]: thanks

**Review by IceTDrinker — APPROVED** [review]
Thanks !

**Pull request merged**